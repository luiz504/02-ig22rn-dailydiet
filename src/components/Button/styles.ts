import { TouchableHighlight } from 'react-native'
import styled from 'styled-components/native'

export const Container = styled(TouchableHighlight)`
  padding: 14px 24px;
  border-radius: 6px;

  justify-content: center;
  align-items: center;
  border-width: 1.5px;
  border-style: solid;
`

export const Content = styled.View`
  flex-direction: row;
  align-items: center;
  gap: 8px;
`
