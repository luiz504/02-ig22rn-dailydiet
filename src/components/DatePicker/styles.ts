import { Pressable } from 'react-native'
import styled from 'styled-components/native'
import { Text } from '../Text'

export const Button = styled(Pressable).attrs(({ theme }) => ({
  android_ripple: { color: theme.colors['black-06%'] },
}))`
  border-radius: 6px;
  border-width: 1.5px;
  border-color: ${({ theme }) => theme.colors['gray-300']};
  padding: 14px;
`
export const Label = styled(Text)``
