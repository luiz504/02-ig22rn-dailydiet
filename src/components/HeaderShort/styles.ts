import { TouchableOpacity } from 'react-native'
import styled from 'styled-components/native'

export const Container = styled.View`
  flex-direction: row;
  padding: 17px 24px 32px 16px;
  align-items: center;
  justify-content: center;
  position: relative;
`

export const BtnReturn = styled(TouchableOpacity)`
  position: absolute;
  left: 24px;
  bottom: 28px;
`
