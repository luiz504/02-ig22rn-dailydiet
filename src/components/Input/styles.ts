import { css, styled } from 'styled-components/native'
import { TextInput as TextInputNative } from 'react-native'
type TextInputProps = {
  isFocused: boolean
}

export const StyledInput = styled(TextInputNative)<TextInputProps>`
  padding: 11px 14px;
  border-radius: 6px;
  border-width: 1.5px;

  font-size: 16px;

  ${({ isFocused, theme }) => css`
    border-color: ${isFocused
      ? theme.colors['gray-600']
      : theme.colors['gray-300']};

    color: ${theme.colors['gray-800']};
    font-family: ${theme.fontFamily.regular};
  `}
`
