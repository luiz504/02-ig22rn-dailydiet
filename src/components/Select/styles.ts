import styled, { css, DefaultTheme } from 'styled-components/native'

import { Text } from '../Text'
import { Pressable } from 'react-native'

export type Variant = 'green' | 'red'

type ContainerProps = {
  variant: Variant
  isSelected: boolean
}

const buttonsVariantStyle: Record<
  Variant,
  (theme: DefaultTheme) => ReturnType<typeof css>
> = {
  green: (theme) => css`
    border-color: ${theme.colors['green-900']};
    background-color: ${theme.colors['green-100']};
  `,
  red: (theme) => css`
    border-color: ${theme.colors['red-900']};
    background-color: ${theme.colors['red-100']};
  `,
}

export const Container = styled(Pressable)<ContainerProps>`
  flex-direction: row;
  align-items: center;
  justify-content: center;

  padding: 16px;
  gap: 8px;
  border-radius: 8px;
  border-width: 1.5px;

  border-color: ${({ theme }) => theme.colors['gray-200']};
  background-color: ${({ theme }) => theme.colors['gray-200']};

  ${({ variant, theme, isSelected }) =>
    isSelected && buttonsVariantStyle[variant](theme)}
`

type DotProps = {
  variant: Variant
}

export const Dot = styled.View<DotProps>`
  width: 8px;
  height: 8px;
  border-radius: 99999px;

  background-color: ${({ theme, variant }) =>
    variant === 'green' ? theme.colors['green-900'] : theme.colors['red-900']};
`

export const Label = styled(Text).attrs({ size: 'sm', weight: 'bold' })``
