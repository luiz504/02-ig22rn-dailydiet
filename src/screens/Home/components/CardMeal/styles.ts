import { TouchableOpacity } from 'react-native'
import styled from 'styled-components/native'

export const Container = styled(TouchableOpacity)`
  padding: 14px 16px 14px 12px;
  border-radius: 6px;
  gap: 12px;
  border-width: 1.5px;
  border-color: ${({ theme }) => theme.colors['gray-300']};

  flex-direction: row;
  align-items: center;
`
export const HR = styled.View`
  background-color: ${({ theme }) => theme.colors['gray-400']};
  width: 1px;
  height: 100%;
`
export const StatusIndicator = styled.View`
  height: 14px;
  width: 14px;
  border-radius: 9999px;
`
