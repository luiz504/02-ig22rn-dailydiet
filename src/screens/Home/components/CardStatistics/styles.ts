import { Pressable } from 'react-native'
import styled from 'styled-components/native'
import ArrowUp from '~/assets/icons/arrow-right-up.svg'
import Animated from 'react-native-reanimated'

export const AnimatedWrapper = styled(Animated.View)`
  margin-top: 12px;
  margin-bottom: 40px;
`

export const Container = styled(Pressable)`
  padding: 20px 16px;
  border-radius: 8px;

  position: relative;
`
export const ContentAnimated = styled(Animated.View)`
  justify-content: center;
  align-items: center;
`

export const ArrowIcon = styled(ArrowUp)`
  position: absolute;
  top: 0px;
  right: 0px;
`
