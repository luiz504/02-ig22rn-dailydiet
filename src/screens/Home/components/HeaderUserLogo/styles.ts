import styled from 'styled-components/native'

export const Container = styled.View`
  padding: 24px 0;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`
export const ProfileWrapper = styled.View`
  height: 40px;
  width: 40px;
  border-radius: 20px;
  border: 2px solid ${({ theme }) => theme.colors['gray-700']};
`
