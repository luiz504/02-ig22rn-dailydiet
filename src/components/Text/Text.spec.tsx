import { render } from '~/utils/test-utils'
import { Text } from '.'
import { theme } from '~/styles'

describe('Text Component', () => {
  const fontSizes = Object.keys(theme.fontSize)
  it('should render without errors with default style', () => {
    const componentSnapshot = render(<Text />).toJSON()

    expect(componentSnapshot).toMatchSnapshot()
  })

  it('should render with props weight = bold', () => {
    const componentSnapshot = render(<Text weight="bold" />).toJSON()

    expect(componentSnapshot).toMatchSnapshot()
  })

  fontSizes.forEach((fontSize) => {
    it(`should be able to integrate theme fontSize "${fontSize}" within the props "size"`, () => {
      const componentSnapshot = render(
        <Text size={fontSize as keyof typeof theme.fontSize} />,
      ).toJSON()

      expect(componentSnapshot).toMatchSnapshot()
    })
  })
})
