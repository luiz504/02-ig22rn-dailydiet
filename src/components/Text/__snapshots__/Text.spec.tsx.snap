// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should be able to integrate theme fontSize "2xl" within the props "size" 1`] = `
<Text
  size="2xl"
  style={
    {
      "fontFamily": "Nunito-Regular",
      "fontSize": 32,
      "lineHeight": 41.6,
    }
  }
/>
`;

exports[`Text Component should be able to integrate theme fontSize "lg" within the props "size" 1`] = `
<Text
  size="lg"
  style={
    {
      "fontFamily": "Nunito-Regular",
      "fontSize": 18,
      "lineHeight": 23.4,
    }
  }
/>
`;

exports[`Text Component should be able to integrate theme fontSize "md" within the props "size" 1`] = `
<Text
  size="md"
  style={
    {
      "fontFamily": "Nunito-Regular",
      "fontSize": 16,
      "lineHeight": 20.8,
    }
  }
/>
`;

exports[`Text Component should be able to integrate theme fontSize "sm" within the props "size" 1`] = `
<Text
  size="sm"
  style={
    {
      "fontFamily": "Nunito-Regular",
      "fontSize": 14,
      "lineHeight": 18.2,
    }
  }
/>
`;

exports[`Text Component should be able to integrate theme fontSize "xl" within the props "size" 1`] = `
<Text
  size="xl"
  style={
    {
      "fontFamily": "Nunito-Regular",
      "fontSize": 24,
      "lineHeight": 31.2,
    }
  }
/>
`;

exports[`Text Component should be able to integrate theme fontSize "xs" within the props "size" 1`] = `
<Text
  size="xs"
  style={
    {
      "fontFamily": "Nunito-Regular",
      "fontSize": 12,
      "lineHeight": 15.6,
    }
  }
/>
`;

exports[`Text Component should render with props weight = bold 1`] = `
<Text
  style={
    {
      "fontFamily": "Nunito-Bold",
      "fontSize": 16,
      "lineHeight": 20.8,
    }
  }
  weight="bold"
/>
`;

exports[`Text Component should render without errors with default style 1`] = `
<Text
  style={
    {
      "fontFamily": "Nunito-Regular",
      "fontSize": 16,
      "lineHeight": 20.8,
    }
  }
/>
`;
